File: Chain of Responsibility/PHP/Response.php
<?php

class Response {
    private int $code;
    private string $message;
    private $data;

    function __construct(int $code, string $message, $data) {
        $this->code = $code;
        $this->message = $message;
        $this->data = $data;
    }

    function get_code() {
        return $this->code;
    }

    function get_message() {
        return $this->message;
    }

    function get_data() {
        return $this->data;
    }

    function describe() {
        return $this->code . " : " . $this->message . "\n"; 
    }
}

File: Chain of Responsibility/Dart/Response.dart
class Response {
  int? _statusCode;
  String? _message;
  int? _data;

  Response(this._statusCode, this._message, this._data);

  int? get getCode => _statusCode;
  String? get getMessage => _message;
  int? get getData => _data;

  String describe() {
    return "$_statusCode : $_message\n";
  }
}
